### PR TITLE
Detect HTTP2 and multiplexing support for CURL at runtime

### PR DESCRIPTION
**Pull request by janvorli**
This change allows portable version of dotnet core to run on versions
of CURL with and without HTTP2 and multiplexing support and still be
able to use those features if running on CURL that supports them.
The compile time checks are now made at runtime and detect the feature
support based on comparison of the current version of CURL to the
well known version number that added that feature support. The version
numbers are listed in docs/libcurl/symbols-in-versions in libcurl source
tree.

### CODE REVIEW

**Comment by janvorli** [comment]
@stephentoub can you take a look please?

**Comment by janvorli** [comment]
FYI: @Petermarcu

File: src/Native/Unix/System.Net.Http.Native/pal_versioninfo.cpp [hunk 1]
```diff
@@ -94,12 +97,9 @@ extern "C" int32_t HttpNative_GetCurlVersionInfo(int32_t* age,
     *age = versionInfo->age;
     *supportsSsl = (versionInfo->features & CURL_VERSION_SSL) == CURL_VERSION_SSL;
     *supportsAutoDecompression = (versionInfo->features & CURL_VERSION_LIBZ) == CURL_VERSION_LIBZ;
-    *supportsHttp2Multiplexing =
-#if HAVE_CURLPIPE_MULTIPLEX
-        (versionInfo->features & CURL_VERSION_HTTP2) == CURL_VERSION_HTTP2;
-#else
-        0;
-#endif
+    *supportsHttp2Multiplexing =
```
Comment (stephentoub) [inline src/Native/Unix/System.Net.Http.Native/pal_versioninfo.cpp]: Actually, I think we can delete the whole HttpNative_GetCurlVersionInfo function at this point... I believe it's dead code, replaced by the individual functions like HttpNative_GetSupportsHttp2Multiplexing.  This is confirmed by the TODO at https://github.com/dotnet/corefx/pull/17904/files#diff-c340184a1ed09fe1e5a2c790d293204cR74 and by a search showing that the only occurrence of HttpNative_GetCurlVersionInfo is this definition.

Comment (janvorli) [inline src/Native/Unix/System.Net.Http.Native/pal_versioninfo.cpp]: Ok, I'll remove it

**Pull request merged**